cpp/src/parquet/schema_test.cc

```diff
@@ -43,26 +43,59 @@ namespace schema {
 
 static inline SchemaElement NewPrimitive(const std::string& name,
                                          FieldRepetitionType::type repetition,
-                                         Type::type type, int id = 0) {
+                                         Type::type type, int field_id = -1) {
   SchemaElement result;
   result.__set_name(name);
   result.__set_repetition_type(repetition);
   result.__set_type(static_cast<format::Type::type>(type));
-
+  if (field_id >= 0) {
+    result.__set_field_id(field_id);
+  }
   return result;
 }
 
 static inline SchemaElement NewGroup(const std::string& name,
                                      FieldRepetitionType::type repetition,
-                                     int num_children, int id = 0) {
+                                     int num_children, int field_id = -1) {
   SchemaElement result;
   result.__set_name(name);
   result.__set_repetition_type(repetition);
   result.__set_num_children(num_children);
 
+  if (field_id >= 0) {
+    result.__set_field_id(field_id);
+  }
+
   return result;
 }
 
+template <typename NodeType>
+static void CheckNodeRoundtrip(const Node& node) {
+  format::SchemaElement serialized;
+  node.ToParquet(&serialized);
+  std::unique_ptr<Node> recovered = NodeType::FromParquet(&serialized);
+  ASSERT_TRUE(node.Equals(recovered.get()))
+      << "Recovered node not equivalent to original node constructed "
+      << "with logical type " << node.logical_type()->ToString() << " got "
+      << recovered->logical_type()->ToString();
+  ASSERT_EQ(recovered->field_id(), node.field_id());
```

Comment:
Why not make `field_id` tested by `Node::Equals`?

Reply:
I did that later, I'll remove this superfluous check